guava/src/com/google/common/base/Predicate.java

```diff
@@ -62,7 +63,7 @@
    *     arguments
    */
   @CanIgnoreReturnValue
-  boolean apply(@Nullable T input);
+  boolean apply(@NonNull T input);
```

Comment:
As per the JavaDoc for the `apply` method, it should throw a `NullPointerException` whenever a `input` is `null`.
>  @throws NullPointerException if {@code input} is null and this predicate does not accept null
>   arguments

If the overriding methods supports `@Nullable` arguments then it should be annotated accordingly. Albeit your suggestions makes more sense, but allowing type T arguments for method apply will not be in accordance with the JavaDocs for the given method. Let me what are your thoughts on this and how should I proceed.
Thanks!!

Reply:
In general, annotations should indicate behavior that is guaranteed to throw no exception.  See this section of the manual: https://checkerframework.org/manual/#annotate-normal-behavior .  This may mean that users sometimes need to suppress a warning, but that's better than a false negative where the checker issues no warning but the code suffers an exception at run time.